Add linting, and fix the lints, in Python (with longer lines)

mcpherrinm opened the pull request:
When reviewing python integration tests, I noticed a number of mistakes that would be instantly caught by a linter. So this adds the ruff linter, along with black for formatting.

You probably want to read the individual commits, as there's a mix of big automatic ones and small real fixes:

* Fix references to unknown names
* Explicitly import from helpers
* E741: Ambiguous variable name
* Duplicated test
* F402: Shadowing import
* E722: Do not use bare except
* Missing argument to .format
* move import to top of file
* Ignore imports not at start
* Add black and ruff linting of python files
* duplicate import
* ruff --fix
* Reformat with black @ 200 lines

This is a second run at #6861 but with line-length set to 200, which avoids some robot-formatting that's a bit harder to read.

aarongable commented on test/v2_integration.py:
```diff
@@ -17,7 +17,25 @@
 import chisel2
-from helpers import *
+
+# from helpers import *
```
Don't leave commented-out import behind

mcpherrinm replied:
Oops!  fixed.

aarongable commented on start.py:
```diff
@@ -14,11 +14,11 @@
 
 if not startservers.install(race_detection=False):
-    raise(Exception("failed to build"))
+    raise (Exception("failed to build"))
```
nit: remove the wrapping parens too, here and a few other places (`grep "raise ("`)

mcpherrinm replied:
Ooof, this one is in a ton of places.  Fixed!

aarongable commented on test/startservers.py:
```diff
@@ -11,110 +11,139 @@
-        8016, None,
-        ('./bin/boulder', 'log-validator', '--config', os.path.join(config_dir, 'log-validator.json')),
-        None),
+    Service("boulder-remoteva-a", 8011, "rva1.service.consul:9097", ("./bin/boulder", "boulder-remoteva", "--config", os.path.join(config_dir, "va-remote-a.json")), None),
```
Still not a huge fan of what it's done here -- these should either all be single lines, or all be one-argument-per-line, not a mixture -- but I don't think there's anything to be done about it. The black linter has decided to be opinionated about fitting everything onto one line when it can, so let's just roll with it.

aarongable commented on test/v2_integration.py:
```diff
@@ -1178,36 +1161,42 @@ def test_new_order_policy_errs():
-            raise(Exception("Order problem detail did not match expected"))
+            raise (Exception("Expected rejectedIdentifier type problem, got {0}".format(e.typ)))
+        if (
+            e.detail != 'Error creating new order :: Cannot issue for "between-addr.in-addr.arpa": The ACME server refuses to '
+            "issue a certificate for this domain name, because it is forbidden by policy (and 1 more problems. "
+            "Refer to sub-problems for more information.)"
```
a) this now has a mixture of single- and double-quotes; and
b) let's not use python's implicit multiline string contatenation. Instead, wrap the RHS of this inequality in parens, and explicitly concatenate the lines, imo.

mcpherrinm replied:
I can explicitly wrap and concat, but black style is to prefer double quotes, but it prefers avoiding escaping more, which is a bit of a problem here.

pgporada commented on start.py:
```diff
@@ -12,13 +12,12 @@
 
 if not startservers.install(race_detection=False):
-    raise(Exception("failed to build"))
+    raise (Exception("failed to build"))
```
```suggestion
    raise Exception("failed to build")
```

mcpherrinm commented on test/chisel2.py:
```diff
@@ -31,60 +26,66 @@
-DIRECTORY_V2 = os.getenv('DIRECTORY_V2', 'http://boulder.service.consul:4001/directory')
-ACCEPTABLE_TOS = os.getenv('ACCEPTABLE_TOS',"https://boulder.service.consul:4431/terms/v7")
-PORT = os.getenv('PORT', '80')
+DIRECTORY_V2 = os.getenv("DIRECTORY_V2", "http://boulder.service.consul:4001/directory")
```
done

mcpherrinm commented:
I've taken a slightly more dramatic approach to trying to tame SERVICES:

I've changed the type, making it into a map, and added some helper functions for constructing the most common cases, which cuts down the amount of boilerplate fairly significantly.

It's fairly straightforward to back it out if we don't think that's the way we want to go, though.

mcpherrinm commented:
The merge conflicts for this PR are meaningful enough that it will be easier to start from scratch, so I'm going to close this for now.
